e2e/pom.xml

```diff
@@ -0,0 +1,141 @@
+                                    <arg value="run" />
+                                    <arg value="-v" />
+                                    <arg value="${basedir}/:/root/:Z" />
+                                    <arg value="node:10" />
+                                    <arg value="su" />
+                                    <arg value="-c" />
+                                    <arg value="cd /root; rm -rf dist node_modules" />
+                                </exec>
+                            </target>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>install-build-lint</id>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <target>
+                                <!-- build tests and test them for lint -->
+                                <exec dir="${basedir}" executable="docker" failonerror="true">
+                                    <arg value="run" />
+                                    <arg value="-v" />
+                                    <arg value="${basedir}/:/root/:Z" />
+                                    <arg value="node:10" />
+                                    <arg value="su" />
+                                    <arg value="-c" />
+                                    <arg value="cd /root/; npm i; npm run tsc; npm run lint" />
+                                </exec>
+                            </target>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <profiles>
+        <profile>
+            <id>publish</id>
+            <activation>
+                <property>
+                    <!-- Run this profile if environment variable NPM_AUTH_TOKEN exists -->
+                    <name>env.NPM_AUTH_TOKEN</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-antrun-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>push</id>
+                                <phase>install</phase>
+                                <goals>
+                                    <goal>run</goal>
+                                </goals>
+                                <configuration>
+                                    <target>
+                                        <!-- set version and tag, publish package -->
+                                        <exec dir="${basedir}" executable="bash" failonerror="true">
+                                            <arg value="docker"/>
```

Comment:
Do we have to build this in a docker container? And does it have to be node:10? why not https://access.redhat.com/containers/?tab=overview#/registry.access.redhat.com/ubi8/nodejs-10 ?

Reply:
Building in container is to make sure all dependencies (npm etc.) are present on the machine (even if somebody from community would do `mvn clean install` on this repo).
As for UBI8 images... They require redhat account to pull, right?  That may not be propriate for community project IMO.

Reply:
UBI images are freely available and redistributable, but would introduce some compatibility testing (e.g. no `yarn` IIRC)

Reply:
@nickboldt I've tried to change the image to `registry.access.redhat.com/ubi8/nodejs-10` but when I simply change the image and leave the rest as is, it want's me to authenticate. But when I remove `su -c` I don't know how can I run the command. Could you help me with that please?

Reply:
In the end we solved this ;-) Good suggestion, @nickboldt @amisevsk . Thx ;-) (Katka will update the PR)